src/catchup/CatchupWork.cpp

```diff
@@ -312,8 +312,12 @@ CatchupWork::getAndMaybeSetHistoryArchiveState()
                     ? CatchupConfiguration::CURRENT
                     : mApp.getHistoryManager().checkpointContainingLedger(
                           mCatchupConfiguration.toLedger());
+            // Set retries to something high in case the current checkpoint
+            // hasn't been published yet
+            size_t retries = mApp.getHistoryManager().getCheckpointFrequency() *
```

Comment:
This seems way too much?
We potentially wait for 8 minutes per retry after the 9th retry, and here you're setting it to 340 retries... so this is basically "infinite".

Right now we use the default of "a few" (5 retries), and "a lot" (32) seems to be way too many already. Maybe the real fix here is to
* change `getRetryDelay` to pick a number between "m" and "m/2+1" for m > 2 (instead of 1...m). That way we have a low/high bound for retry timeout a little more sane.
* use a retry count of 9 for `GetHistoryArchiveStateWork` here (that will be a bit less than 512 seconds as low bound, high bound is 1024 seconds)